beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/model/HousekeepingEntity.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.core.model;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+import com.expediagroup.beekeeper.core.monitoring.Taggable;
+
+public interface HousekeepingEntity extends Taggable {
+
+  String getLifecycleType();
+
+  void setLifecycleType(String lifecycleType);
```

Comment:
Do you really need all those setters, you'll always have to build the concrete class right, this should probably just be a subset of the getters that are shared in concrete subclasses.

Reply:
True. I've just changed them to getters :-)